pandas/tools/tile.py

```diff
@@ -185,6 +185,13 @@ def qcut(x, q, labels=None, retbins=False, precision=3, duplicates='raise'):
     x, dtype = _coerce_to_type(x)
 
     if is_integer(q):
+        if x.size == 0:
+            raise ValueError('Cannot qcut empty array')
+
+        rng = (nanops.nanmin(x), nanops.nanmax(x))
+        if rng[0] == rng[1] and q == 1:
```

Comment:
Sure, I can do that but this seems cleaner than increasing the edges by 0.1% for this specific case only (q=1 and identical values). That seems fine with pd.cut but not with pd.qcut, considering that we might be returning those bins (it's not hidden to the user).

Also, I thought that with a little extra work we could make 'allow' a top level pd.qcut option. Compared to 'drop' this option would maintain the total number of quantiles (some of them empty in the case of duplicated bin edges).

What do you think? If you don't like it I'll go with the 0.1% edges increment.